fixed Broken UI for ValueSet table view

Ruthwik000 opened the pull request:
## Proposed Changes

Fixes #13482

- Added overflow-x-auto to table container div in ValueSetList component
- Replaced overflow-hidden with overflow-x-auto to enable horizontal scrolling
- Reduced table cell padding from px-6 to px-3 to minimize table width
- Removed unnecessary whitespace-nowrap classes from table cells
- Added max-width constraints and truncation to prevent excessive column widths
- Prevents table content from being cut off when it exceeds viewport width on large screens

<img width="2533" height="1194" alt="image" src="https://github.com/user-attachments/assets/da10b4ae-6aa1-4bfc-be88-65e57e586207" />


Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is placed in I18n files.
- [ ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Valueset table now shows full descriptions by default (no per-row truncation).

* **Style**
  * Removed per-item "Read more" buttons in the table; descriptions use natural wrapping.
  * Simplified expandable text behavior: inline "Read more" toggles are used where needed, improving readability and layout.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AdityaJ2305 reviewed (commented):
https://github.com/user-attachments/assets/8ce20764-348f-49a3-a1ae-687700bd4216


Issue still persist for screen size shown in the video attached 👆 (unnecessary scroll)

AdityaJ2305 commented on src/components/ValueSet/ValueSetList.tsx:
```diff
@@ -201,102 +201,104 @@ const RenderTable = ({
-                        <TooltipTrigger className="flex w-full">
-                          <div className="text-sm font-medium text-gray-900 truncate">
+    <div className="hidden lg:block rounded-lg bg-white shadow-sm">
+      <div className="overflow-x-auto">
```
Instead we should check why overflowing is happening at first place ? 🤔

Ruthwik000 replied:
@AdityaJ2305  What does it mean first place, can you clearly comment your thoughts ..
And here clear explaintion 
Root cause: The table was forced to be wider than the screen because:
min-w-full - Forces table to take full width even if content doesn't need it
whitespace-nowrap - Prevents text from wrapping, making cells stretch horizontally
Large padding (px-6) - Adds unnecessary width to each cell
No column width limits - Cells could grow infinitely wide

and here how i fixed it : 
 Removed min-w-full → Changed to w-full table-auto
Removed all whitespace-nowrap → Let text wrap naturally
Reduced padding px-6 → px-3
Added column widths → Controlled how much space each column takes
Changed container overflow-hidden → overflow-x-auto

AdityaJ2305 reviewed (commented):
https://github.com/user-attachments/assets/2510b1b7-7aa9-4d7b-a581-6c8f926c5dd9



Issue still persist

Ruthwik000 commented:
Hi @Jacobjeevan @rithviknishad
I fixed the conflicts long back but there is merge updated kindly review and merge 
thankyou

AdityaJ2305 reviewed (commented):
<img width="1110" height="89" alt="Screenshot 2025-09-09 at 9 02 41 PM" src="https://github.com/user-attachments/assets/6b4f3a4b-5669-4ab3-b94c-6adb1d0fcb6a" />

description is being clipped out

Ruthwik000 commented:
@AdityaJ2305  fixed everything , kindly review

![WhatsApp Image 2025-09-09 at 22 22 41_78fceb66](https://github.com/user-attachments/assets/ca152854-c514-4aa4-99ad-986c4d6a67d1)

AdityaJ2305 commented on src/components/ValueSet/ValueSetList.tsx:
```diff
@@ -148,15 +122,8 @@ const RenderCard = ({
                     {t("description")}
                   </h3>
-                  <div className="max-w-md text-sm text-gray-900 break-words whitespace-normal">
-                    <ExpandableText>
-                      <ExpandableTextContent>
-                        {valueset.description}
-                      </ExpandableTextContent>
-                      <ExpandableTextExpandButton>
-                        {t("read_more")}
-                      </ExpandableTextExpandButton>
-                    </ExpandableText>
```
I don't think we want ExpandableText to be removed from here 

Moreover, current changes doesn't make it to perfect ui 

![Screenshot 2025-09-10 at 4 12 36 PM](https://github.com/user-attachments/assets/46028020-07a6-4cc0-b4af-ac5f03ad39b2)

Ruthwik000 commented:
Problem
• Text overflowed horizontally out of table cells when "Read more" clicked

Solution
• ExpandableText Component: Complete rewrite with React Context, 30-char truncation, inline "Read more" button • ValueSetList Component: Added align-top to table cell, removed separate button component • Result: Text now expands vertically within table cell instead of overflowing horizontally

Key Changes
• Replaced complex overflow detection with simple character limit • Made "Read more" button inline with text • Used original Button styling (variant="secondary", size="xs") • Table cells now expand vertically to show full content

https://github.com/user-attachments/assets/ea3e9e5b-3917-4251-92a9-5c5ad4e9e4d5

Ruthwik000 commented:
@AdityaJ2305  kindly review now fixed the text overflowing text from table  fixed to Reliable vertical expansion without horizontal overflow

nihal467 commented:
LGTM

rithviknishad commented on src/components/ui/expandable-text.tsx:
let's not modify this component...
